src/resdk/resources/data.py

```diff
@@ -354,20 +357,48 @@ def download(
             "{}/{}".format(self.id, fname)
             for fname in self.files(file_name, field_name)
         ]
+        file_names = [fname for fname in self.files(file_name, field_name)]
 
-        # Only applies if downloading a single file
-        custom_file_names = None
-        if custom_file_name:
-            if file_name or field_name:
-                custom_file_names = [custom_file_name] * len(files)
-            else:
-                raise ValueError(
-                    "Setting a custom file name is not supported "
-                    "without specifying file name or field name."
-                )
+        self.resolwe._download_files(files=files, download_dir=download_dir)
+
+        return file_names
+
+    def download_and_rename(
+        self,
+        custom_file_name: str,
+        field_name: Optional[str] = None,
+        file_name: Optional[str] = None,
+        download_dir: Optional[str] = None,
+    ):
```

Comment:
Just curious: is the use case when you want to download just a single filename really used a lot?

Reply:
Yes, I almost exclusively download just one file from a data object.